HipercowApi/Tools/MetricsUpdateService.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace HipercowApi.Tools
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Hpc.Scheduler;
+    using Microsoft.Hpc.Scheduler.Properties;
+
+    /// <summary>
+    /// The implementation of IMetrics.
+    /// </summary>
+    /// <remarks>
+    /// Initializes a new instance of the <see cref="MetricsUpdateService"/> class.
+    /// </remarks>
+    /// <param name="clusterHandleCache">
+    /// Handle cache so we can query the headnode.
+    /// </param>
+    public class MetricsUpdateService(
+        IClusterHandleCache clusterHandleCache) : BackgroundService
+    {
+        private readonly IClusterHandleCache clusterHandleCache = clusterHandleCache;
+        private readonly Dictionary<string, Dictionary<string, dynamic>> userJobs = [];
+
+        /// <summary>
+        /// Calculate core hours for a Finished or Running job.
+        /// </summary>
+        /// <param name="job">PropertyRow for the job, returned from the
+        /// ISchedulerJobEnumerator.</param>
+        /// <param name="state">HPC JobState for the job.
+        /// Since I deal with states separately, I don't query this back
+        /// from the cluster, so it's not in the job property, to keep that
+        /// query as light as possible. Perhaps pedantic.</param>
+        /// <returns>The number of core hours, as a float.</returns>
+        internal static float GetCoreHours(PropertyRow job, JobState state)
+        {
```

Comment:
maybe some spacing will help the readability of this function

Reply:
Yeah. I am a little bit annoyed with stylecop and might start to ignore it as here it clearly makes things worse. Particularly, casts not being allowed to have a space before the object seems nasty to me...